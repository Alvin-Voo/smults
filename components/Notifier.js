import React, { Component } from 'react';
import PropTypes from 'prop-types';

export const ERROR = 'ERROR';
export const MESSAGE = 'MESSAGE';

export default class Notifier extends Component {
  displayMessage() {
    const { type, message } = this.props;
    switch (type) {
      case ERROR:
        return (
          <div className="error">
            <p>{ message }</p>
            <p>
          Sorry, something is broken. Take a screenshot and notify
              <a href="https://steemit.com/@alvinvoo" target="_blank" rel="noopener noreferrer">
                {' '}
            @alvinvoo
              </a>
            </p>
          </div>
        );
      case MESSAGE:
        return (
          <p>{ message }</p>
        );
      default:
        return <div />;
    }
  }

  render() {
    return this.displayMessage();
  }
}

Notifier.propTypes = {
  type: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};
