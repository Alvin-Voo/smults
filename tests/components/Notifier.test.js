import React from 'react';
import { shallow } from 'enzyme';
import Notifier, { ERROR, MESSAGE } from '../../components/Notifier';

test('should return an empty <DIV /> by default', () => {
  const wrapper = shallow(<Notifier type="" message="" />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ERROR display correctly', () => {
  const errorMessage = 'some error message';
  const wrapper = shallow(<Notifier type={ERROR} message={errorMessage} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render MESSAGE display correctly', () => {
  const message = 'some message';
  const wrapper = shallow(<Notifier type={MESSAGE} message={message} />);
  expect(wrapper).toMatchSnapshot();
});
