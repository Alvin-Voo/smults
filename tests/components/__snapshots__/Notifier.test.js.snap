// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ERROR display correctly 1`] = `
<div
  className="error"
>
  <p>
    some error message
  </p>
  <p>
    Sorry, something is broken. Take a screenshot and notify
    <a
      href="https://steemit.com/@alvinvoo"
      rel="noopener noreferrer"
      target="_blank"
    >
       
      @alvinvoo
    </a>
  </p>
</div>
`;

exports[`should render MESSAGE display correctly 1`] = `
<p>
  some message
</p>
`;

exports[`should return an empty <DIV /> by default 1`] = `<div />`;
